clang/lib/3C/ConstraintResolver.cpp

```diff
@@ -164,7 +164,7 @@ CSetBkeyPair ConstraintResolver::getExprConstraintVars(Expr *E) {
     E = E->IgnoreParens();
 
     // Non-pointer (int, char, etc.) types have a special base PVConstraint.
-    if (TypE->isRecordType() || TypE->isArithmeticType()) {
+    if (TypE->isRecordType() || TypE->isArithmeticType() || TypE->isVectorType()) {
```

Comment:
Since this condition had to change it three places it might be worth pulling it out into a util function.

Reply:
Done